src/query/service/src/evaluator/scalar.rs

```diff
@@ -61,14 +63,49 @@ impl Evaluator {
                 Ok(EvalNode::Function { func, args })
             }
             Scalar::FunctionCall(func) => {
-                let args: Vec<EvalNode> = func
+                let eval_args: Vec<EvalNode> = func
                     .arguments
                     .iter()
                     .map(Self::eval_scalar)
                     .collect::<Result<_>>()?;
+
+                // special case for in function
+                let name_lower = func.func_name.to_lowercase();
+                if name_lower.as_str() == "in" || name_lower.as_str() == "not_in" {
+                    if let EvalNode::Constant {
+                        value: DataValue::Struct(vs),
+                        ..
+                    } = &eval_args[1]
+                    {
+                        let func = if name_lower.as_str() == "not_in" {
+                            in_evalutor::create_by_values::<true>(
+                                func.arg_types[0].clone(),
+                                vs.clone(),
+                            )
+                        } else {
+                            in_evalutor::create_by_values::<false>(
+                                func.arg_types[0].clone(),
+                                vs.clone(),
+                            )
+                        }?;
+
+                        return Ok(EvalNode::Function {
+                            func,
+                            args: vec![eval_args[0].clone()],
+                        });
+                    } else {
+                        return Err(ErrorCode::SyntaxException(
+                            "IN expression must have a literal array or subquery as the second argument",
+                        ));
```

Comment:
We only support IN a literal array or subquery?

BTW, IN subquery seems always be converted to JOIN.

Reply:
> BTW, IN subquery seems always be converted to JOIN.

Yes, so in function only covered literal array.

Reply:
Will we allow non-literal array in the future? Many other DBs support it.